ecl/hqlcpp/hqlresource.cpp

```diff
@@ -3091,15 +3095,37 @@ bool ResourcerInfo::expandRatherThanSpill(bool noteOtherSpills)
             return false;
 
         node_operator op = expr->getOperator();
+
+        if (isKnownLikelihood(filterLikelihood))
+        {
+            switch(op)
+            {
+            case no_table:
+            case no_inlinetable:
+            case no_getresult:
+            case no_temptable:
+            case no_rows:
+            case no_xmlproject:
+            case no_workunit_dataset:
+            case no_keyindex:
+            case no_newkeyindex:
+            case no_datasetfromrow:
+            case no_hqlproject:
+            case no_select:
+                if (filterLikelihood >= options->likelihoodThresholdToSpill)
+                    return true;
```

Comment:
This can only be returned from sources - e.g., table, inlinetable, retresult.  The cases that return it below.